pkgs/development/tools/parsing/tree-sitter/grammar.nix

```diff
@@ -30,7 +30,11 @@ stdenv.mkDerivation {
     if [ ! -f "$scanner_cc" ]; then
       scanner_cc=""
     fi
-    $CC -I$src/src/ -shared -o parser -Os $src/src/parser.c $scanner_cc -lstdc++
+    scanner_c="$src/src/scanner.c"
+    if [ ! -f "$scanner_c" ]; then
+      scanner_c=""
+    fi
```

Comment:
```suggestion
    if [ -f "$scanner_c" ]; then
      scanner_c="$src/src/scanner.c"
    else
      scanner_c=""
    fi
```

Reply:
`scanner_c` would be undefined at the point where you `[ -f "$scanner_c" ]`, so it will always take the "empty $scanner_c" path. It's fine as-is.